ts/Dashboards/Components/EditableOptions.ts

```diff
@@ -98,5 +57,93 @@ class EditableOptions {
         return this.component.options.editableOptions;
     }
 }
+namespace EditableOptions {
+
+    /**
+     * Configuration for a single option in editable options. If type is
+     * `nested` the options are rendered in the accordion menu, with rest of the
+     * options defined in the detailed options.
+     */
+    export interface Configuration {
+        /**
+         * Name of the option which will be displayed on the label.
+         */
+        name: string;
+        /**
+         * Type of the editable element.
+         */
+        type: ElementType;
+        /**
+         * Detailed options that should be included in the accordion menu.
+         * Available for `nested` type.
+         */
+        nestedOptions?: Array<NestedOptions>
+        /**
+         * Relative path to the option, that should be changed in the component.
+         * eg: `['chart', 'title', 'text']`
+         */
+        propertyPath?: Array<string>
+        /**
+         * Items that should be included in the select element.
+         */
+        selectOptions?: Array<SelectOptions>;
+    }
+
+    /**
+     * Options of the single option in the select dropdown.
+     */
+    export interface SelectOptions {
+        /**
+         * Name of the item that should be displayed.
+         */
+        name: string;
+        /**
+         * URL of the icon that should be displayed. It is concatenated with
+         * `iconURLPrefix` option.
+         */
+        iconURL?: string;
+    }
+
+    /**
+     * Type of the input to be displayed.
+     */
+    export type ElementType =
+        | 'input'
+        | 'text'
+        | 'textarea'
+        | 'toggle'
+        | 'select'
+        | 'nested';
+
+    /**
+     * Configuration for a single option in detailed options.
+     */
+    export interface NestedOptions {
+        /**
+         * Name of the option that should be displayed.
+         */
+        name: string;
+        /**
+         * whether the option should have a toggle to be enabled or disabled.
+         */
+        allowEnabled?: boolean;
```

Comment:
Just realized:
- upper case
- what exactly does this option do?

Reply:
This option adds a toggle to a header option. 
<img width="326" alt="Screenshot 2023-05-12 at 08 45 35" src="https://github.com/highcharts/highcharts/assets/26306846/4e2dde94-c4d4-4992-a204-a9eb54849e08">

Reply:
Hm.. how about "showToggle" name then? 🤔

Reply:
Great name :D